apps/prairielearn/python/prairielearn/core.py

```diff
@@ -1062,17 +1048,18 @@ def string_from_2darray(
     return result
 
 
-def string_from_number_sigfig(
-    a: complex | np.complex64 | np.complex128 | numbers.Number, digits: int = 2
-) -> str:
+def string_from_number_sigfig(a: _NumPyScalarType, digits: int = 2) -> str:
```

Comment:
I believe this new type is correct, please check though.

Reply:
I added `number.Number` as an acceptable input type for all these functions in eaf4c1c. It's not clear what "logic" around `numbers.Number` is needed, can you elaborate?

Reply:
The logic I referred to was the `isinstance numbers.Number` check, which I know understand to be correct.